apps/files/src/components/PublicLinksSidebar/EditPublicLink.vue

```diff
@@ -0,0 +1,292 @@
+<template>
+  <div class="oc-files-edit-public-link oc-files-file-link-form">
+    <form @submit.prevent>
+      <transition enter-active-class="uk-animation-slide-top-small" leave-active-class="uk-animation-slide-top-small uk-animation-reverse"
+                  name="custom-classes-transition">
+        <div class="uk-alert-danger" uk-alert v-if="errors">
+          <a class="uk-alert-close" uk-close/>
+          <p v-text="errors"/>
+        </div>
+      </transition>
+      <div class="uk-margin">
+        <label class="oc-label"><span v-translate>Name your link</span>:</label>
+        <input class="uk-input" id="oc-files-file-link-name" v-model="name"/>
+      </div>
+      <oc-grid childWidth="1-1" gutter="small">
+        <roles-select :roles="$_roles" :selectedRole="$_selectedRole" @roleSelected="$_selectRole" mode="file-links"/>
+        <div class="uk-text-muted" v-if="$_selectedRole_description" v-text="$_selectedRole_description"></div>
+      </oc-grid>
+      <h4 class="uk-margin-medium-top uk-heading-divider" v-translate>
+        Security settings
+      </h4>
+      <div class="uk-margin uk-grid-small uk-flex uk-flex-middle" uk-grid>
+        <div class="uk-width-1-1 uk-width-2-5@m" v-if="$_expirationDate">
+          <label class="oc-label" for="oc-files-file-link-expire-date">
+            <span v-translate>Expiration date</span>:<em class="uk-margin-small-left" v-if="$_expirationDate.enforced">(<span v-translate>required</span>)</em>
+          </label>
+          <div class="uk-position-relative">
+            <oc-datepicker :class="{ 'uk-form-danger': !$_expirationIsValid }" :date="expireDate" :key="'oc-datepicker-' + expireDate"
```

Comment:
The placeholder didn't seem to work in my local test, the field appears completely empty. Please check.

It is also likely that the ODS component doesn't have a placeholder. If that's the case please submit a PR to https://github.com/owncloud/owncloud-design-system/

Reply:
Good catch. In fact, `OcDatepicker` has no placeholder prop. Will add a PR there.

Reply:
https://github.com/owncloud/owncloud-design-system/pull/618